### DISCUSSION TITLE
Auto3DSeg and transforms

### DISCUSSION THREAD

**Discussion by adrianvallsc** [Project-MONAI/MONAI#6584] [Q&A] [answered]
I am running Auto3DSeg on a public [dataset](https://isles22.grand-challenge.org). The main problem is dimensions of labels and images do not fit and are not homogeneous. 

With standard MONAI pipelines I was using transforms such as ResampletoMatch to make them homogeneous. When I run Auto3DSeg as I could not find how to customize the transforms it fails:

RuntimeError: Caught RuntimeError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 141, in apply_transform
    return _apply_transform(transform, data, unpack_items, lazy, overrides, log_stats)
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 98, in _apply_transform
    return transform(data, lazy=lazy) if isinstance(transform, LazyTrait) else transform(data)
  File "/usr/local/lib/python3.10/dist-packages/monai/apps/auto3dseg/transforms.py", line 82, in __call__
    raise ValueError(
ValueError: The label shape torch.Size([112, 112, 72]) is different from the source shape torch.Size([352, 352, 29]) /content/drive/MyDrive/segmentation/LABELS/sub-strokecase0242_ses-0001_msk.nii.gz.

**Comment by KumoLiu** [comment]
Hi @adrianvallsc, from the error message, I think you have a different size between the image and the label. You may need to check this data "/content/drive/MyDrive/segmentation/LABELS/sub-strokecase0242_ses-0001_msk.nii.gz".
Thanks!

**Comment by adrianvallsc** [comment]
Yes, I guessed this was the reason. My question was wether it was possible to apply custom transforms to Auto3DSeg run

**Comment by KumoLiu** [comment, marked as answer]
Hi @adrianvallsc, I think the error was before training, it was in the `data_analyzer` step, so adding your custom transforms may not solve this error. You may need to check the data first. :)
https://github.com/Project-MONAI/MONAI/blob/575954cbe0a27ed076806ca3f6f5024114f08cad/monai/apps/auto3dseg/data_analyzer.py#L299

Thanks!
